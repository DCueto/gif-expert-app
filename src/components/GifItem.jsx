import PropTypes from 'prop-types';

export const GifItem = ({ title, path }) => {

  return (
    <div className='card'>
      <img src={path} alt={title} />
      <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired
}
