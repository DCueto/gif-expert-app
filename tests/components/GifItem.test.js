import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas para el componente de GifItem', () => {

  const title = 'One Punch Man Deepanshu GIF';
  const path = 'https://media0.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=99c9bfceiwn7b4eh0iirynl87muyn7fwoxvpr4b9hr8oiul6&ep=v1_gifs_search&rid=giphy.gif&ct=g';

  test('Coincide con el snapshot', () => {

    const { container } = render( <GifItem 
      title={title}
      path={path} 
      /> );

    expect( container ).toMatchSnapshot();
    
  });

  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

    render( <GifItem title={title} path={path} /> );
    // screen.debug();

    // expect( screen.getByRole('img').src ).toBe( path );
    // expect( screen.getByRole('img').alt ).toBe( title );

    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( path );
    expect( alt ).toBe( title );

  });

  test('Debe de mostrar el tÃ­tulo en el componente', () => {

    render( <GifItem title={title} path={path} /> );

    expect( screen.getByText( title ) ).toBeTruthy();
    
  });
  
});