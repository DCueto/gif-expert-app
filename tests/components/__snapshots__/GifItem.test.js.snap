// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente de GifItem Coincide con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man Deepanshu GIF"
      src="https://media0.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=99c9bfceiwn7b4eh0iirynl87muyn7fwoxvpr4b9hr8oiul6&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man Deepanshu GIF
    </p>
  </div>
</div>
`;
